### PR TITLE
Serialize odometry data

### PR DESCRIPTION
**Pull request by ojura**
Replaces #550.

### CODE REVIEW

**Comment by ojura** [comment]
By the way, this PR is almost line-for-line identical to #548, `/s/IMU/Odometry`. And a similar one will be  forthcoming for fixed frame pose data in 3D. Perhaps this could be somehow simplified? Introducing all these getters for the optimization problem and the pose graph is noise, IMHO.

It would be nice if we had all these `MapByTime` objects in a single structure in the optimization problem, and if we could just ask this structure to serialize itself from `MapBuilder`.

File: cartographer/mapping/map_builder.cc [hunk 1]
```diff
@@ -193,7 +193,19 @@ void MapBuilder::SerializeState(io::ProtoStreamWriter* const writer) {
       }
     }
   }
-  // TODO(whess): Serialize additional sensor data: odometry.
```
Comment (wohe) [inline cartographer/mapping/map_builder.cc]: Retain the `TODO`? There is still fixed frame pose data missing.

File: cartographer/mapping/proto/serialization.proto [hunk 2]
```diff
@@ -37,9 +37,14 @@ message ImuData {
...
+message OdometryData {
+  int32 trajectory_id = 1;
+  sensor.proto.OdometryData odometry_data = 2;
+}
+
 message SerializedData {
   Submap submap = 1;
   Node node = 2;
   ImuData imu_data = 3;
-  // TODO(whess): Add odometry.
```
Comment (wohe) [inline cartographer/mapping/proto/serialization.proto]: Similar here: retain for fixed frame pose data?

**Comment by ojura** [comment]
Updated. I also forgot to expose `AddOdometryData` in the SPG base class in `mapping/sparse_pose_graph.h`, so I added it.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 2.

**Pull request merged**